### PR TITLE
expression: implement vectorized evaluation for `builtinSubTimeStringNullSig`

### PR DESCRIPTION
**Pull request by SilvaXiang**
### What problem does this PR solve? 
implement vectorized evaluation for `builtinSubTimeStringNullSig `

### What is changed and how it works?
44% faster than before:

    BenchmarkVectorizedBuiltinTimeFunc/builtinSubTimeStringNullSig-VecBuiltinFunc-4                   343850              3766 ns/op               0 B/op          0 allocs/op
    BenchmarkVectorizedBuiltinTimeFunc/builtinSubTimeStringNullSig-NonVecBuiltinFunc-4                170829              6800 ns/op               0 B/op          0 allocs/op
    
    

    

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/generator/time_vec.go [hunk 1]
```diff
@@ -6,6 +6,7 @@
 //
 //     http://www.apache.org/licenses/LICENSE-2.0
 //
+
```
Comment (francis0407) [inline expression/generator/time_vec.go]: please revert this change.

File: expression/builtin_time_vec_test.go [hunk 2]
```diff
@@ -124,7 +124,20 @@ var vecBuiltinTimeCases = map[string][]vecExprBenchCase{
...
+	ast.SubTime: {
+		// builtinSubTimeStringNullSig
+		{
+			retEvalType:   types.ETString,
+			childrenTypes: []types.EvalType{types.ETDatetime, types.ETDatetime},
+
+			childrenFieldTypes: []*types.FieldType{types.NewFieldType(mysql.TypeDate), types.NewFieldType(mysql.TypeDatetime)},
+
+			geners: []dataGenerator{
+				gener{defaultGener{eType: types.ETDatetime, nullRation: 0.2}},
```
Comment (francis0407) [inline expression/builtin_time_vec_test.go]: It seems we don't need this.

File: expression/builtin_time_vec_test.go [hunk 3]
```diff
@@ -124,7 +124,20 @@ var vecBuiltinTimeCases = map[string][]vecExprBenchCase{
 	ast.TimestampLiteral: {},
 	ast.SubDate:          {},
 	ast.AddDate:          {},
-	ast.SubTime:          {},
+	ast.SubTime: {
+		// builtinSubTimeStringNullSig
+		{
+			retEvalType:   types.ETString,
+			childrenTypes: []types.EvalType{types.ETDatetime, types.ETDatetime},
+
```
Comment (francis0407) [inline expression/builtin_time_vec_test.go]: We don't blank lines here.

File: expression/builtin_time_vec.go [hunk 4]
```diff
@@ -2006,11 +2006,19 @@ func (b *builtinSubDateDurationStringSig) vecEvalDuration(input *chunk.Chunk, re
...
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+
+	result.ReserveString(n)
+	for i := 0; i < n; i++ {
+		result.AppendNull()
+	}
+
+	return nil
+
```
Comment (francis0407) [inline expression/builtin_time_vec.go]: please remove this blank line.

File: expression/builtin_time_vec_test.go (hunk 2 above)
Comment (francis0407) [reply expression/builtin_time_vec_test.go]: Do we really need to specify a gener here? Can we use the default gener?

Comment (SilvaXiang) [reply expression/builtin_time_vec_test.go]: So I have changed it to nil.

Comment (francis0407) [reply expression/builtin_time_vec_test.go]: Even `nil` is unnecessary. You can just remove `geners:` here.

**Comment by SilvaXiang** [comment]
@sre-bot How can I merge successfully?
